pep-0646.rst

```diff
@@ -414,6 +355,100 @@ Normal ``TypeVar`` instances can also be prefixed and/or suffixed:
+allow unpacking a tuple type. As we shall see, this also enables a
+number of interesting features.
+
+
+Unpacking a Concrete Tuple Type
+'''''''''''''''''''''''''''''''
+
+Unpacking a concrete tuple type is analogous to unpacking a tuple of
+values at runtime. ``Tuple[int, *Tuple[bool, bool], str]`` is
+equivalent to ``Tuple[int, bool, bool, str]``.
+
+Unpacking an Unbounded Tuple Type
+'''''''''''''''''''''''''''''''''
+
+Unpacking an unbounded tuple preserves the unbounded tuple as it is.
+That is, ``*Tuple[int, ...]`` remains ``*Tuple[int, ...]``; there's no
+simpler form. This enables us to specify types such as ``Tuple[int,
+*Tuple[str, ...], str]`` - a tuple type where the first element is
+guaranteed to be of type ``int``, the last element is guaranteed to be
+of type ``str``, and the elements in the middle are zero or more
+elements of type ``str``. Note that ``Tuple[*Tuple[int, ...]]`` is
+equivalent to ``Tuple[int, ...]``.
+
+Unpacking unbounded tuples is also useful in function signatures where
+we don't care about the exact elements and don't want to define an
+unnecessary ``TypeVarTuple``:
+
+::
+
+    def process_batch_channels(
+        x: Array[Batch, *Tuple[Any, ...], Channels]
+    ) -> None:
+        ...
+
+
+    x: Array[Batch, Height, Width, Channels]
+    process_batch_channels(x)  # OK
+    y: Array[Batch, Channels]
+    process_batch_channels(y)  # OK
+    z: Array[Batch]
+    process_batch_channels(z)  # Error: Expected Channels.
+
+
+We can also pass a ``*Tuple[int, ...]`` wherever a ``*Ts`` is
+expected. This is useful when we have particularly dynamic code and
+cannot state the precise number of dimensions or the precise types for
+each of the dimensions. In those cases, we can smoothly fall back to
+an unbounded tuple:
+
+::
+
+    y: Array[*Tuple[Any, ...]] = read_from_file()
+
+    def expect_variadic_array(
+        x: Array[Batch, *Shape]
+    ) -> None: ...
+
+    expect_variadic_array(y)  # OK
+
+    def expect_precise_array(
+        x: Array[Batch, Height, Width, Channels]
+    ) -> None: ...
+
+    expect_precise_array(y)  # OK
+
+``Array[*Tuple[Any, ...]]`` stands for an array with an arbitrary
+number of dimensions of type ``Any``. This means that, in the call to
+``expect_variadic_array``, ``Batch`` is bound to ``Any`` and ``Shape``
+is bound to ``Tuple[Any, ...]``. In the call to
+``expect_precise_array``, the variables ``Batch``, ``Height``,
+``Width``, and ``Channels`` are all bound to ``Any``.
+
+This allows users to handle dynamic code gracefully while still
+explicitly marking the code as unsafe (by using ``y: Array[*Tuple[Any,
+...]]``).  Otherwise, users would face noisy errors from the type
+checker every time they tried to use the variable ``y``, which would
+hinder them when migrating a legacy code base to use ``TypeVarTuple``.
+
+Multiple Unpackings in a Tuple: Not Allowed
+'''''''''''''''''''''''''''''''''''''''''''
+
+As with ``TypeVarTuple``, only one unpacking may appear in a tuple:
```

Comment:
Explain why? (Maybe put the explanation in the corresponding section for TVT.) IIRC, the reason is that of we have e.g. [int, str, str, str, str], there are multiple solutions for Ts and Ts2.